code/game/machinery/recycle_vendor.dm

```diff
@@ -94,120 +84,208 @@
 		return
 
 	eject_stored_item()
 
 
-/obj/machinery/recycle_vendor/proc/eject_stored_item()
+/obj/machinery/amerecycler/proc/eject_stored_item(obj/stored_item_object)
 	if(stored_item_object)
 		stored_item_object.forceMove(loc)
-		stored_item_object = null
-		flick("recycle_vend", src)
-		update_icon()
-
-
-/obj/machinery/recycle_vendor/proc/eject_stored_materials(material_type)
-	if(material_type)
-		var/material/M = get_material_by_name(material_type)
-		new M.stack_type(loc, materials_stored[material_type])
-		materials_stored -= material_type
+		if(stored_item_object in saleworthy_items)
+			saleworthy_items.Remove(stored_item_object)
 	else
-		for(var/i in materials_stored)
-			var/material/M = get_material_by_name(i)
-			new M.stack_type(loc, materials_stored[i])
-		materials_stored.Cut()
+		for(var/obj/item/item in saleworthy_items)
+			item.forceMove(loc)
+		saleworthy_items.Cut()
 
 	flick("recycle_vend", src)
 	update_icon()
+	
 
+/obj/machinery/amerecycler/emag_act(remaining_charges, mob/user, emag_source)
+	. = ..()
+	sales_paused = TRUE
+	to_chat(user, SPAN_NOTICE("[src]'s display flashes red."))
+	update_icon()
 
-/obj/machinery/recycle_vendor/proc/evaluate_stored_item()
-	stored_item_value = 0
-	stored_item_fluff = "Material composition:"
-	stored_item_materials = stored_item_object.get_matter()
-
-	if(stored_item_object.contents.len)	// Storage items are too resource intensive to check (populate_contents() means we have to create new instances of every object within the initial object)
-		stored_item_fluff += "<br>ERROR Storage Item Nesting Detected."
-		return
-
-	for(var/i in stored_item_materials)
-		if(i in materials_supported)
-			if(i in materials_allowed)
-				var/material/M = get_material_by_name(i)
-				var/obj/item/stack/material/S = new M.stack_type(null, stored_item_materials[i])
-				stored_item_value += round(S.get_item_cost(), 2) / 2
-				stored_item_fluff += "<br>[i] - [stored_item_materials[i]] units, worth [round(S.get_item_cost(), 2) / 2] credits."
-			else
-				stored_item_fluff += "<br>Payouts for [i] suspended by Aster Guild representative."
-		else // Bay leftover materials
-			stored_item_fluff += "<br>[i] recycling is not supported."
-
-	stored_item_fluff += "<br>Payout total: [stored_item_value] credits."
-
-
-/obj/machinery/recycle_vendor/proc/recycle_and_pay()
-	// That account doesn't exist on initialize, hence
-	// cannot be assigned to pre-mapped vendor in sane manner
-	if(!merchants_pocket)
-		merchants_pocket = department_accounts[DEPARTMENT_GUILD]
+/obj/machinery/amerecycler/proc/evaluate_stored_item(obj/evaluated)
+	if(!silo)
+		return FALSE // if there is no silo, it is not stored
+
+// don't lie in the comments.
+
+	var/valueofitem = 0
+	var/list/intermediary = evaluated.get_matter()
+	var/matsinitem = intermediary?.Copy()
+	for(var/obj/O in evaluated.GetAllContents()) // can now recycle empty shells to get at the contents
+		if(length(O.get_matter()))
+			matsinitem += O.get_matter()
```

Comment:
This properly adds the value of each key of `O.get_matter()` to the value of the corresponding keys in `matsinitem`?

Reply:
testing said yes.